packages/ra-ui-materialui/src/theme/useTheme.ts

```diff
@@ -23,7 +24,13 @@ export type ThemeSetter = (theme: ThemeType | RaThemeOptions) => void;
 export const useTheme = (
     type?: ThemeType | RaThemeOptions
 ): [ThemeType | RaThemeOptions, ThemeSetter] => {
+    const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)', {
+        noSsr: true,
+    });
     // FIXME: remove legacy mode in v5, and remove the RaThemeOptions type
-    const [theme, setter] = useStore<ThemeType | RaThemeOptions>('theme', type);
+    const [theme, setter] = useStore<ThemeType | RaThemeOptions>(
```

Comment:
`useTheme` returns the theme in the store even if it's not available in the application. Shouldn't we use `ThemesContext` to check if a `darkTheme` is defined?

Reply:
Good catch! FIxed